Adyen/CheckoutConfiguration/CheckoutBaseCallbacks.swift

```diff
@@ -27,3 +27,49 @@ package protocol CheckoutBaseCallbacks {
     
     var onComplete: CheckoutSuccessHandler? { get set }
 }
+
+/// Data model to contain relevant parts of `/payments` and `/payment/details` call responses.
+public struct CheckoutPaymentsResponse: Decodable, Sendable {
+    
+    // TODO: should it be string or enum?
+    public let resultCode: String
+    
+    public let action: Action?
+    
+    public let order: PartialPaymentOrder?
+    
+    public init(from decoder: any Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.resultCode = try container.decode(String.self, forKey: .resultCode)
+        self.action = try container.decodeIfPresent(Action.self, forKey: .action)
+        self.order = try container.decodeIfPresent(PartialPaymentOrder.self, forKey: .order)
+    }
+    
+    private enum CodingKeys: CodingKey {
+        case resultCode
+        case action
+        case order
+    }
+}
+
+/// Data model that contains information regarding the status of a payment.
+public struct CheckoutResult {
+    
+    // TODO: string or enum?
+    public let resultCode: String
+    
+    package init(resultCode: String) {
+        self.resultCode = resultCode
+    }
+}
+
+/// Wrapper type to contain checkout related errors.
+// TODO: if not needed, can remove?
+public struct CheckoutError: Error {
```

Comment:
Does it make sense to merge both into Result<CheckoutResult, CheckoutError>?

Reply:
result code exists regardless of error. it's not a case of one or the other so we can't create a `Result` with these two. 
also cc: @neelSharma12 relevant to your comment